lib/util/device-service-details.ts

```diff
@@ -110,26 +119,37 @@ export const generateCurrentServiceDetails = (
 		getSingleInstallSummary(gd),
 	);
 
-	const device = omit(rawDevice, [
-		'image_install',
-		'gateway_download',
-	]) as DeviceWithServiceDetails;
-
-	device.current_services = mapValues(
-		groupBy(installs, 'service_name'),
-		serviceContainers => {
-			return serviceContainers
-				.map(container => {
-					return omit(container, 'service_name') as Omit<
-						typeof container,
-						'service_name'
-					>;
-				})
-				.sort((a, b) => {
-					return b.install_date.localeCompare(a.install_date);
-				});
-		},
+	// prefer over omit for performance reasons
+	delete rawDevice.image_install;
+	delete rawDevice.gateway_download;
+
+	const device = (rawDevice as Device) as DeviceWithServiceDetails;
+
+	// Essentially a groupBy(installs, 'service_name')
+	// but try making it a bit faster for the sake of large fleets
+	const currentServicesGroupedByName = installs.reduce<
+		Record<string, CurrentService[]>
+	>((acc, container) => {
+		let serviceContainerGroup = acc[container.service_name];
+		if (!serviceContainerGroup) {
+			serviceContainerGroup = acc[container.service_name] = [];
+		}
+
+		// remove the extra property that we added for the grouping
+		delete container.service_name;
+		serviceContainerGroup.push(container);
+
+		return acc;
+	}, {});
+
+	// TODO: change to Object.values() in a later major release
+	Object.keys(currentServicesGroupedByName).forEach(serviceName =>
```

Comment:
For performance it's probably better to do
```suggestion
	for (const serviceName in currentServicesGroupedByName) {
```
rather than creating the intermediary array which will later need to be GCed